.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  # https://blog.rust-lang.org/2022/06/22/sparse-registry-testing.html
+  CARGO_UNSTABLE_SPARSE_REGISTRY: "true"
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    container: devkitpro/devkitarm
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+
+      - name: Setup default Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          components: clippy, rustfmt, rust-src
+          profile: minimal
+          toolchain: nightly
+          default: true
+
+      - name: Install build tools for host
+        run: sudo apt-get install -y build-essential
+
+      - name: Install cargo-3ds
+        uses: actions-rs/cargo@v1
+        with:
+          command: install
+          # TODO: this should probably just be a released version from crates.io
+          # once cargo-3ds gets published somewhere...
+          args: >-
+            --git https://github.com/rust3ds/cargo-3ds
+            --rev 913645665391ea715bc96910bda35f98a4536a6e
+
+      - name: Check formatting
+        run: cargo fmt --all --verbose -- --check
+
+      # For some reason, `cargo clippy` doesn't seem to work properly with
+      # -Zbuild-std (and thus with `cargo 3ds`). Maybe would be helped by
+      # https://github.com/rust3ds/cargo-3ds/pull/21 ?
+      - name: Cargo check
+        run: cargo 3ds check --color=always --workspace --verbose --all-targets
+        env:
+          RUSTFLAGS:
+            --deny=warnings
+
+  # TODO: it would be nice to actually build 3dsx for examples/tests, etc.
+  # and run it somehow, but exactly how remains to be seen. Also, we probably
+  # need the std::thread PR to land before a lot of the examples are runnable.
```

Comment:
I tried using `--exclude` to ignore the thread-related examples, but it doesn't seem to work with examples :confused:.
https://doc.rust-lang.org/cargo/commands/cargo-build.html#package-selection

Reply:
Pushed a change with a feature to disable building the thread examples unless `--all-features` (or the explicit `std-threads` feature) is enabled.